tests/aws/services/secretsmanager/test_secretsmanager.py

```diff
@@ -533,58 +608,65 @@ def test_rotate_secret_with_lambda_success(
         Tests secret rotation via a lambda function.
         Parametrization ensures we test the default behavior which is an immediate rotation.
         """
-        cre_res = create_secret(
-            Name=secret_name,
-            SecretString="my_secret",
-            Description="testing rotation of secrets",
-        )
-
-        sm_snapshot.add_transformer(
-            sm_snapshot.transform.key_value("RotationLambdaARN", "lambda-arn")
-        )
-        sm_snapshot.add_transformers_list(
-            sm_snapshot.transform.secretsmanager_secret_id_arn(cre_res, 0)
+        rotation_config = {
+            "RotationRules": {"AutomaticallyAfterDays": 1},
+        }
+        if rotate_immediately:
+            rotation_config["RotateImmediately"] = rotate_immediately
+        initial_secret_version, function_arn = self._setup_rotation_secret(
+            sm_snapshot, secret_name, create_secret, create_lambda_function, aws_client
         )
-
-        function_name = f"s-{short_uid()}"
-        function_arn = create_lambda_function(
-            handler_file=TEST_LAMBDA_ROTATE_SECRET,
-            func_name=function_name,
-            runtime=Runtime.python3_12,
-        )["CreateFunctionResponse"]["FunctionArn"]
-
-        aws_client.lambda_.add_permission(
-            FunctionName=function_name,
-            StatementId="secretsManagerPermission",
-            Action="lambda:InvokeFunction",
-            Principal="secretsmanager.amazonaws.com",
+        self._assert_after_rotate_secret_with_lambda_success(
+            sm_snapshot,
+            secret_name,
+            initial_secret_version,
+            aws_client,
+            function_arn=function_arn,
+            rotation_config=rotation_config,
         )
 
-        rotation_kwargs = {}
-        if rotate_immediately is not None:
-            rotation_kwargs["RotateImmediately"] = rotate_immediately
-        rot_res = aws_client.secretsmanager.rotate_secret(
-            SecretId=secret_name,
-            RotationLambdaARN=function_arn,
-            RotationRules={
-                "AutomaticallyAfterDays": 1,
-            },
-            **rotation_kwargs,
+    @pytest.mark.parametrize("set_rotation_on_second_run", [False, True])
+    @markers.snapshot.skip_snapshot_verify(
+        paths=["$..VersionIdsToStages", "$..Versions", "$..VersionId"]
+    )
+    @markers.aws.validated
+    def test_rotate_secret_multiple_times_with_lambda_success(
+        self,
+        sm_snapshot,
+        secret_name,
+        create_secret,
+        create_lambda_function,
+        aws_client,
+        set_rotation_on_second_run,
+    ):
```

Comment:
It would be worth testing - what happens if the `function_arn` is not provided for the first time rotation.